### PR TITLE
Add CallToAction screen & Modify FigureCard Component

### PR DESCRIPTION
**Pull request by TanMar23**
## Description
This PR is intended to resolve https://app.clubhouse.io/lanahub/story/4002/calltoactionscreen-implementation-in-the-ui-library

"We saw many times this screen repeated across different repos. This screen is just about a title + description + image + button, so it would be nice to make it generic and configurable to avoid code to be repeated each time we have to implement a screen similar than this one."

## Testing
- npm run lint: **PASS**
- npm run test: **PASS**
- Tested using Storybook

## Screenshots/Captures
CallToActionScreen

## Checks
- [ ] Requires documentation update
- [x] Requires lib version update

## Versioning impact
- [x] **MAJOR** incompatible API changes.
- [ ] **MINOR** adds functionality in a backwards-compatible manner.
- [ ] **PATCH** backwards-compatible bug fixes.

### CODE REVIEW

File: src/components/CallToActionScreen/CallToActionScreen.js [hunk 1]
```diff
@@ -24,22 +25,22 @@ const props = {
     type: String,
     default: '',
   },
-  closeButtonText: {
+  ButtonText: {
```
Comment (jasonadt) [inline src/components/CallToActionScreen/CallToActionScreen.js:28]: (camelCase) please rename this prop with standard camelCase to: `buttonText`

File: src/components/CallToActionScreen/CallToActionScreen.stories.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    device: {
+      default: select('Simulated Mobile Device', [...availableDevices], availableDevices[0]),
+    },
+    title: {
+      default: text('Title', 'Example Title'),
+    },
+    description: {
+      default: text('Description', 'Some description'),
+    },
+    ButtonText: {
```
Comment (jasonadt) [inline src/components/CallToActionScreen/CallToActionScreen.stories.js:31]: Same here, should be: `buttonText`

File: src/components/CallToActionScreen/CallToActionScreen.stories.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    onClick: action('Click!'),
+  },
+  template: `
+  <div style="margin: 10px 50px 10px 50px;">
+    <h2><strong>CallToActionScreen:</strong>&nbsp;This can be used as a generic and configurable screen that has the following: title, description, image and button.</h2>
+    <hr>
+    <StorybookMobileDeviceSimulator :device="device">
+      <CallToActionScreen :title="title"
+                          :description="description"
+                          :ButtonText="ButtonText"
```
Comment (jasonadt) [inline src/components/CallToActionScreen/CallToActionScreen.stories.js:45]: Please always use `kebab-case` for markup props. Should be: 

```js
:button-text="buttonText"
```

File: src/components/CallToActionScreen/CallToActionScreen.js (hunk 1 above)
Comment (jantocv) [inline src/components/CallToActionScreen/CallToActionScreen.js:28]: Lowercase first ;) 

```suggestion
  buttonText: {
```

File: src/components/CallToActionScreen/CallToActionScreen.vue [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  <Screen>
+    <Wrapper modal>
+      <Heading class="title">{{ title }}</Heading>
+      <TextParagraph> {{ description }}</TextParagraph>
+    </Wrapper>
+    <Wrapper>
+      <slot/>
+    </Wrapper>
+    <div class="button-container">
+      <WrappedButton :data-testid="dataTestId" @click="onClick"> {{ ButtonText }} </WrappedButton>
```
Comment (jasonadt) [inline src/components/CallToActionScreen/CallToActionScreen.vue:11]: Should be: `buttonText`

**Review by jasonadt — CHANGES_REQUESTED** [review]
Nice work! Just a few small changes before I'll approve.

File: src/components/CallToActionScreen/CallToActionScreen.vue [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<template>
+  <Screen>
+    <Wrapper modal>
+      <Heading class="title">{{ title }}</Heading>
+      <TextParagraph> {{ description }}</TextParagraph>
+    </Wrapper>
+    <Wrapper>
+      <slot/>
+    </Wrapper>
```
Comment (jantocv) [inline src/components/CallToActionScreen/CallToActionScreen.vue:9]: Adding a ScrollWrapper here:

```suggestion
<ScrollWrapper>
    <Wrapper modal>
      <Heading class="title">{{ title }}</Heading>
      <TextParagraph> {{ description }}</TextParagraph>
    </Wrapper>
    <Wrapper>
      <slot/>
    </Wrapper>
</ScrollWrapper>
```

You can remove the `<div class="button-container">` and let the `<WrappedButton>` to be alone ;)

File: src/components/CallToActionScreen/CallToActionScreen.scss [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+  text-align: center;
+  padding: 32px 16px 8px;
+}
+
+.button-container {
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-end;
+}
```
Comment (jantocv) [inline src/components/CallToActionScreen/CallToActionScreen.scss:11]: If you add the `<ScrollScreen>` you will not need this ;)

**Review by jantocv — CHANGES_REQUESTED** [review]
Good job!!! 🚀  Please check my suggestions, specially the `ScrollScreen` one, because it will help to have a scrollable content in the `slot` part and not to hide the Button when we scroll it.

**Review by jasonadt — DISMISSED** [review]
👍  great work! 🎉

**Review by jantocv — APPROVED** [review]
I was waiting for this PR :) well done 🚀

**Review by jasonadt — APPROVED** [review]
👍  🎉  ship it!

**Pull request merged**